Story 2

FelixLetheren opened the pull request.

FelixLetheren committed:
Added icon on index that links to new fish entry page.

FelixLetheren committed:
Created basic HTML form

FelixLetheren committed:
Corrected an alt on newFish.php

FelixLetheren committed:
added label to aggression selector.

FelixLetheren committed:
Closed img tag
ran through validator and resolved errors and warnings
changed return image to a clearer icon.

FelixLetheren committed:
Merge pull request #7 from FelixLetheren/S2AT2-BasicForm

S2 at2 basic form

FelixLetheren committed:
Made working function for inserting data from form into database.

FelixLetheren committed:
made length entry a number field and unit of measure clearer.

FelixLetheren committed:
Made $_POST checking check that ALL fields and filled.

FelixLetheren committed:
Made requested changes

FelixLetheren committed:
Merge pull request #8 from FelixLetheren/S2AT4-InsertSLQQuery

Made working function for inserting data from form into database.

FelixLetheren committed:
Added confirmation of success or error.

FelixLetheren committed:
Made boxes larger and added formatting

FelixLetheren committed:
Moved confirmation message to the middle of the page.

FelixLetheren committed:
Changed form to columns

FelixLetheren committed:
Changed confirmation message to only show when POST is set

FelixLetheren committed:
Made suggested formatting changes and made a new unit test for malformed data.

FelixLetheren committed:
Added default image placeholder

FelixLetheren committed:
Added more form formatting.

FelixLetheren committed:
Merge pull request #9 from FelixLetheren/S2BT1-comfirmationOfSucsessOrError

Added confirmation of success or error.

FelixLetheren committed:
Merge branch 'story-2' into S2CT1-Formatting

FelixLetheren committed:
Merge pull request #10 from FelixLetheren/S2CT1-Formatting

S2 ct1 formatting

sjr-mayden commented on php-files/functions.php:
```diff
@@ -56,7 +56,41 @@ function displayFish(array $fishFromDB):string
+ * Takes a boolean and returns 2 different response message strings if true or false
+ */
+function inputConfirmation(bool $dataEntryResult): string
+{
```
OOP code should have `{` on a new line, not functions

sjr-mayden commented on newFish.php:
```diff
@@ -0,0 +1,57 @@
+        <input class="field" type="text" placeholder="Name" name="name" required>
+        <input class="field" type="text" placeholder="Species" name="species" required>
+        <input class="field" type="number" placeholder="Length(cm)" name="length" required>
+        <h3>Aggression:</h3>
```
This should be a label because selects need labels

sjr-mayden commented on test/FunctionsTest.php:
```diff
@@ -62,11 +61,40 @@ public function testDisplayFish_errorsWithUnsetName()
-    public function testDisplayFish_throwsTypeErrorWithWrongInputType(){
+    public function testDisplayFish_throwsTypeErrorWithWrongInputType()
+    {
         $examplefish = 23;
```
camel case all variables please

sjr-mayden commented on newFish.php:
```diff
@@ -0,0 +1,57 @@
+    </a>
+</div>
+<div class="content-container">
+    <?php if(isset($_POST)) echo $message;?>
```
Probably want to be more specific here, relying on a super global variable's existence is likely to go bad

sjr-mayden commented on php-files/functions.php:
```diff
@@ -56,7 +56,41 @@ function displayFish(array $fishFromDB):string
+ * Takes an array and inserts the data into the database
+ */
+function insertEntryIntoDB($dataBase, array $fishArray): bool
+{
```
Fails silently, input should be validated before the call to insertEntryIntoDB happens in newFish.php

FelixLetheren committed:
Moved {'s onto same line as function in functions.php

FelixLetheren committed:
Changed h3 to a label

FelixLetheren committed:
Changed test functions to camelcase

FelixLetheren committed:
Changed isset for confirmation message to look for entries in post rather than POST itself.

FelixLetheren committed:
Added bit of authentication for inputs

FelixLetheren committed:
Got error message to read after entering too many chars.

FelixLetheren committed:
Added verification to limit characters on entry to avoid silent errors.

FelixLetheren committed:
Alterations to limits
Raised max characters from 10 to 15
Reduced max lenth from 999999999999999 to 300 (it's fish we're talking about here)
Made aggression score out of 5 clearer
Changed length to be stored as an INT instead of BIGINT (bit overkill)

FelixLetheren committed:
Fixed Unit Tetst

FelixLetheren committed:
Fixed other Unit Test

FelixLetheren committed:
Added authentication for ensuring length is a possitive length.
